parastell/invessel_build.py

```diff
@@ -501,32 +508,325 @@ def extract_solids_and_mat_tags(self):
+            },
+            {
+                "type": "number",
+                "name": "Parameters/Create surfaces guaranteed to be parametrizable",
+                "values": [0],
+                "choices": [0, 1]
+            }
+            ]"""
+        )
+
+        for component in components:
+            volume_id = self.radial_build.radial_build[component]["vol_id"]
+
+            vtk_path = str(Path(f"volume_{volume_id}_tmp").with_suffix(".vtk"))
+            self.dag_model.volumes_by_id[volume_id].to_vtk(vtk_path)
+
+            gmsh.merge(vtk_path)
+
+        angle = gmsh.onelab.getNumber(
+            "Parameters/Angle for surface detection"
+        )[0]
+        includeBoundary = True
+        forceParameterizablePatches = gmsh.onelab.getNumber(
+            "Parameters/Create surfaces guaranteed to be parametrizable"
+        )[0]
+        curveAngle = 180.0
+
+        gmsh.model.mesh.classifySurfaces(
+            np.deg2rad(angle),
+            includeBoundary,
+            forceParameterizablePatches,
+            np.deg2rad(curveAngle),
+        )
+
+        gmsh.model.mesh.create_geometry()
+
+        surfaces = gmsh.model.getEntities(2)
+        surface_tags = [s[1] for s in surfaces]
+        surface_loop = gmsh.model.geo.addSurfaceLoop(surface_tags)
+        gmsh.model.geo.addVolume([surface_loop])
+
+        gmsh.model.geo.synchronize()
+
+    def _gmsh_from_cadquery(self, components):
+        """Adds CadQuery geometry to Gmsh instance.
+        (Internal function not intended to be called externally)
+
+        Arguments:
+            components (array of str): array containing the names of the
+                in-vessel components to be meshed.
         """
         for component in components:
-            vol_id = import_step_cubit(component, import_dir)
-            mesh_volume_auto_factor([vol_id], mesh_size=mesh_size)
-            export_mesh_cubit(
-                filename=component,
-                export_dir=export_dir,
-                delete_upon_export=True,
+            gmsh.model.occ.importShapesNativePointer(
+                self.Components[component].wrapped._address()
             )
 
+        gmsh.model.occ.synchronize()
+
+    def export_mesh_gmsh(self, filename, export_dir=""):
+        """Exports a tetrahedral mesh of in-vessel component volumes in H5M
+        format via Gmsh and MOAB.
+
+        Arguments:
+            filename (str): name of H5M output file.
+            export_dir (str): directory to which to export the h5m output file
+                (defaults to empty string).
+        """
+        self._logger.info(
+            "Exporting mesh H5M file for in-vessel component(s)..."
+        )
+
+        vtk_path = str(Path(export_dir) / Path(filename).with_suffix(".vtk"))
+        moab_path = Path(export_dir) / Path(filename).with_suffix(".h5m")
+
+        gmsh.write(vtk_path)
+
+        gmsh.clear()
+        gmsh.finalize()
+
+        os.system(f"mbconvert {vtk_path} {moab_path}")
```

Comment:
can't we use pymoab to do this? I have a vague recollection that there was some reason we couldn't but not sure?

Reply:
I also remember that there's a reason we went this route in the past, but I also don't remember why.